interp/src/structures/environment.rs

```diff
@@ -108,127 +110,179 @@ impl InterpreterState {
     fn make_primitive(
         prim_name: &ir::Id,
         params: &ir::Binding,
-        cell_name: Option<&ir::Id>,
+        cell_name: &ir::Id,
         mems: &Option<MemoryMap>,
+        qin_name: &ComponentQIN,
     ) -> InterpreterResult<Box<dyn Primitive>> {
+        let cell_qin = QualifiedInstanceName::new(qin_name, cell_name);
         Ok(match prim_name.as_ref() {
-            "std_const" => Box::new(combinational::StdConst::new(params)),
+            "std_const" => {
+                Box::new(combinational::StdConst::new(params, cell_qin.as_id()))
```

Comment:
Seems like the `.as_id()` call can be lifted up in the definition of `cell_qin`.